addons/advanced_fatigue/functions/fnc_getMetabolicCosts.sqf

```diff
@@ -16,12 +16,14 @@
  * Public: No
  */
 #include "script_component.hpp"
-params ["_unit", "_velocity"];
+params ["_unit", "_movementSpeed"];
 
 private _gearMass = ((_unit getVariable [QEGVAR(movement,totalLoad), loadAbs _unit]) / 22.046) * GVAR(loadFactor);
 
-private _terrainAngle = asin (1 - ((surfaceNormal getPosASL _unit) select 2));
-private _terrainGradient = (_terrainAngle / 45 min 1) * 5 * GVAR(terrainGradientFactor);
+private _surfaceNormal = surfaceNormal (getPosASL _unit);
```

Comment:
`getPosWorld`. Same X,Y; but skips the Z offset from model center to LandContact.